src/components/Charts/ChartCaseDensity.tsx

```diff
@@ -120,6 +123,9 @@ const ChartCaseDensity: FunctionComponent<{
     />
   );
 
+  const dateScale = getDateScale(minDate, currDate, chartWidth);
```

Comment:
The AxisBottom components needs to use the same scale than the data points for them to be aligned, so using `xScale` here will solve the misalignment problem.

The `xScale` is a bit more complicated to calculate than usual, because we don't know the end date of the scale

![image](https://user-images.githubusercontent.com/114084/106963922-2abdc580-66f6-11eb-8139-3131432a32a0.png)

We know that the startDate should be at `0`, and that `currDate` should go to `innerWidth - labelsWidth` (`520` px), but we don't know the `endDate` of the scale. To calculate that, we create a scale, calculate `endDate` and then adjust the scale

```ts
const xScale = scaleUtc({ 
  domain: ["2020-03-01", "2021-02-04"], 
  range: [0, 520]
});

// xScale("2020-03-01") = 0
// xScale("2021-02-04") = 520
```

now we need to calculate `endDate` such that `xScale(endDate) = 600`, fortunately, `xScale.invert` can make the calculation for us:

```ts
endDate = xScale.invert(600);
```

and then we adjust the domain

```ts
xScale.domain([startDate, endDate]);
```
This logic is in [components/Charts/utils.ts#L144](https://github.com/covid-projections/covid-projections/blob/develop/src/components/Charts/utils.ts#L144)